### PR TITLE
[DEV-3162] create /invoice screen

### PR DESCRIPTION
**Pull request by lapatric**
- [DEV-3162] create /invoice screen

[DEV-3162]: https://dfxswiss.atlassian.net/browse/DEV-3162?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by TaprootFreak** [comment]
nice nice nice

File: src/screens/invoice.screen.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    const formattedDate = new Intl.DateTimeFormat('de-DE').format(nowPlusOneYear);
+    const invoiceIdIsNumber = !isNaN(Number(data.invoiceId));
+
+    const callback = url(
+      baseUrl,
+      new URLSearchParams({
+        [invoiceIdIsNumber ? 'routeId' : 'route']: data.invoiceId,
+        amount: data.amount.toString(),
+        currency: data.currency.name,
+        message: data.recipient,
```
Comment (davidleomay) [inline src/screens/invoice.screen.tsx:50]: Did you mix up invoice ID and recipient? 🤔  (looking at the ticket)

File: src/screens/invoice.screen.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        date: formattedDate,
+      }),
+    );
+
+    navigate(callback);
+  }
```
Comment (davidleomay) [inline src/screens/invoice.screen.tsx:56]: There should be an option to copy the URL

File: src/screens/invoice.screen.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    const recipient = urlParams.get('recipient');
+    if (recipient) setValue('recipient', recipient);
+    setUrlParams(new URLSearchParams());
+  }, []);
+
+  useEffect(() => {
+    const baseUrl = '/pl';
+
+    const nextYearDate = addYears(new Date(), 1);
+    const formattedDate = format(nextYearDate, 'yyyy-MM-dd');
```
Comment (davidleomay) [inline src/screens/invoice.screen.tsx:59]: You should be able to use `nextYearDate.toISOString()`

**Review by davidleomay — APPROVED** [review]

**Pull request merged**